activerecord/test/cases/base_test.rb

```diff
@@ -1122,15 +1123,14 @@ def test_no_limit_offset
     end
   end
 
-  def test_find_last
-    last  = Developer.last
-    assert_equal last, Developer.all.merge!(:order => 'id desc').first
-  end
-
   def test_last
     assert_equal Developer.all.merge!(:order => 'id desc').first, Developer.last
   end
 
+  def test_first_and_last_are_equal_without_primary_key
+    assert_equal Season.first, Season.last
```

Comment:
In reading the [code](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/finder_methods.rb#L166), I saw the `last`method checks already if the primary key is present thus I deduced the last author wanted to handle this case.

Currently, without my modification:
```ruby
Price.last    #=> ActiveRecord::StatementInvalid
Price.last(1) #=> [#<Price …>]
```